solutions/cf-862B.mdx

```diff
@@ -0,0 +1,122 @@
+---
+id: cf-862B
+source: CF
+title: Bipartiteness
+author: Ryan Chou
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/54590) 
+
+## Explanation
+Given the number of colored and uncolored nodes, the maximum number of edges that can exist is $c \cdot uc$ where $c$ represents the number of colored nodes, and $uc$ represents the number of uncolored nodes. 
+
+This is because an edge can be drawn from each colored node to every other uncolored node. However, we already have $n - 1$ edges, so the answer is $(c \cdot uc) - (n - 1)$.
+
+To find the number of colored and uncolored nodes, we can run a [DFS](https://usaco.guide/silver/dfs), incrementing the number of colored and uncolored nodes as we traverse the graph.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N + M)$
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class Bipartiteness {
+	public static ArrayList<Integer> adj[];
+	
+	// counters for colored and uncolored nodes
+	public static long[] cnt = {0, 0};
+
+	public static void main(String[] args) throws IOException{
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		PrintWriter out = new PrintWriter(System.out);
+
+		int n = Integer.parseInt(br.readLine());
+
+		// +1 for 1-indexed nodes
+		adj = new ArrayList[n + 1];
+		
+		for (int i = 0; i <= n; i++) adj[i] = new ArrayList<Integer>();
+
+		StringTokenizer s;
+		for (int i = 0; i < n - 1; i++) {
+			s = new StringTokenizer(br.readLine());
+			int u = Integer.parseInt(s.nextToken());
+			int v = Integer.parseInt(s.nextToken());
+			
+			adj[u].add(v);
+			adj[v].add(u);
+		}
+		
+		dfs(1, 0, false);
+		
+		long ans = (cnt[0] * cnt[1]) - (n - 1); 
+		
+		out.println(ans);
+		out.close();
+	}
+
+	public static void dfs(int node, int parent, boolean color) {
+		cnt[color ? 1 : 0]++;
+
+		for (int u : adj[node]) {
+			// if the adjacent node isn't the parent,
+			// visit that node with the opposite color
+			if (u != parent) {
+				dfs(u, node, !color);
+			}
+		}
+	}
+}
+```
+</JavaSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <vector>
+using namespace std;
+
+const int MAX_N = 100000;
+
+// +1 for 1-indexed nodes
```

Comment:
i'd ask you to make it 0-indexed but you made it so explicit and so nice that i'm just going to leave these be